cg/apps/demultiplex/sample_sheet/sample_sheet_validator.py

```diff
@@ -0,0 +1,172 @@
+"""Module with validator classes for the sample sheet."""
+
+import logging
+from pathlib import Path
+from typing import Type
+
+from cg.apps.demultiplex.sample_sheet.override_cycles_validator import OverrideCyclesValidator
+from cg.apps.demultiplex.sample_sheet.read_sample_sheet import (
+    get_flow_cell_samples_from_content,
+    get_raw_samples,
+    get_sample_type,
+    validate_samples_unique_per_lane,
+)
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample, FlowCellSampleBCLConvert
+from cg.apps.demultiplex.sample_sheet.sample_sheet_models import SampleSheet
+from cg.constants.constants import FileFormat
+from cg.constants.demultiplexing import NAME_TO_INDEX_SETTINGS, SampleSheetBCLConvertSections
+from cg.exc import OverrideCyclesError, SampleSheetError
+from cg.io.controller import ReadFile
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetValidator:
+    """Class for validating the content of a sample sheet."""
+
+    def __init__(self):
+        """Instantiate the class with a sample sheet file path or sample sheet content."""
+        self.content: list[list[str]] | None = None
+        self.sample_type: Type[FlowCellSample] | None = None
+        self.read1_cycles: int | None = None
+        self.read2_cycles: int | None = None
+        self.index1_cycles: int | None = None
+        self.index2_cycles: int | None = None
+        self.is_index2_reverse_complement: bool | None = None
```

Comment:
Design problem: don't pass instance specific data to a class which performs an action

Reply:
Also design consideration: can this validation be encapsulated in the pydantic models instead?

Reply:
Added refactoring of the validation logic to the backlog https://github.com/Clinical-Genomics/cg/issues/2963

Reply:
same as above, discussed in person, left for another PR